### PR TITLE
feat: convert github blob URLs to raw

### PR DESCRIPTION
**Pull request by kanadgupta**
(let me know what you think of this pattern, i'm not in this repo much :sweat-nervous:)

### CODE REVIEW

File: src/lib/utils.ts [hunk 1]
```diff
@@ -13,6 +13,16 @@ export function isBuffer(obj: any) {
   );
 }
 
+/**
+ * Converts GitHub blob URLs to raw URLs
+ */
+export function normalizeUrl(url: string) {
```
Comment (erunion) [inline src/lib/utils.ts:19]:
```suggestion
export function normalizeURL(url: string) {
```

File: src/index.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ export default class OASNormalize {
         return resolve(this.file.toString());
 
       case 'url':
-        const resp = await fetch(this.file).then(res => res.text());
+        const resp = await fetch(utils.normalizeUrl(this.file)).then(res => res.text());
```
Comment (erunion) [inline src/index.ts:73]:
```suggestion
        const resp = await fetch(utils.normalizeURL(this.file)).then(res => res.text());
```

File: src/lib/utils.ts (hunk 1 above)
Comment (kanadgupta) [inline src/lib/utils.ts:19]: I kept this function name pretty generic in case we wanted it to do more in the future!

File: src/lib/utils.ts [hunk 3]
```diff
@@ -13,6 +13,16 @@ export function isBuffer(obj: any) {
...
+  if (url.startsWith('https://github.com/')) {
```
Comment (erunion) [inline src/lib/utils.ts:20]: Should we also look for `/blob/`?

Comment (kanadgupta) [reply src/lib/utils.ts]: I feel like if a URL starts with `https://github.com/` and doesn't contain `/blob/`, it's probably going to fail anyways? But doesn't hurt to have so added in https://github.com/readmeio/oas-normalize/pull/236/commits/baf785b76b5c8bc23eed23d96b03f4430eb9ab19

**Review by erunion — APPROVED** [review]

**Pull request merged**